From f698001202f0e19eb8eb9cc38e9cea2ea8b5d9ea Mon Sep 17 00:00:00 2001
From: Alexander Courtis <alex@courtis.org>
Date: Sat, 20 Apr 2019 22:01:56 +1000
Subject: [PATCH] unfloat visible and optionally apply layout

---
 unfloatvisible.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)
 create mode 100644 unfloatvisible.c

diff --git a/unfloatvisible.c b/unfloatvisible.c
new file mode 100644
index 0000000..ff9b36a
--- /dev/null
+++ b/unfloatvisible.c
@@ -0,0 +1,14 @@
+void
+unfloatvisible(const Arg *arg)
+{
+    Client *c;
+
+    for (c = selmon->clients; c; c = c->next)
+        if (ISVISIBLE(c) && c->isfloating)
+            c->isfloating = c->isfixed;
+
+    if (arg && arg->v)
+        setlayout(arg);
+    else
+        arrange(selmon);
+}
-- 
2.21.0

